hazelcast/src/main/java/com/hazelcast/nio/tcp/WriteHandler.java

```diff
@@ -172,7 +172,7 @@ private void unschedule() {
         // So the outputBuffer is empty, so we are going to unschedule ourselves.
         scheduled.set(false);
 
-        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty()) {
+        if (writeQueue.isEmpty() && urgentWriteQueue.isEmpty() && currentPacket == null) {
```

Comment:
So, problem happens when a packet doesn't fit into buffer and there are no other pending packets in the queue. When both packets queues are empty, we are removing write interest immediately, ignoring partially written packet. And that packet's remaining part is written only when another packet is enqueued and write interest is set. 

Right? Anything wrong in my understanding?